src/containerapp/azext_containerapp/tests/latest/test_containerapp_function.py

```diff
@@ -0,0 +1,483 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+import os
+import time
+import base64
+import unittest
+
+from azure.cli.command_modules.containerapp._utils import format_location
+from unittest import mock
+from azure.cli.core.azclierror import ValidationError, CLIInternalError
+
+from azure.cli.testsdk.scenario_tests import AllowLargeResponse, live_only
+from azure.cli.testsdk import (ScenarioTest, ResourceGroupPreparer, JMESPathCheck, JMESPathCheckNotExists, JMESPathCheckExists, live_only, StorageAccountPreparer, LogAnalyticsWorkspacePreparer)
+from azure.mgmt.core.tools import parse_resource_id
+
+from azext_containerapp.tests.latest.common import (write_test_file, clean_up_test_file)
+from .common import TEST_LOCATION, STAGE_LOCATION
+from .custom_preparers import SubnetPreparer
+from .utils import create_containerapp_env, prepare_containerapp_env_for_app_e2e_tests, prepare_containerapp_env_v1_for_app_e2e_tests
+
+TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+
+
+class ContainerappFunctionTests(ScenarioTest):
+    def __init__(self, *arg, **kwargs):
+        super().__init__(*arg, random_config_dir=True, **kwargs)
+    
+
+    @AllowLargeResponse(8192)
+    @ResourceGroupPreparer(location="northcentralus")
+    def test_containerapp_function_list_show_basic(self, resource_group):
+        """Test basic function list functionality with various scenarios"""
+        location = "northcentralusstage"
```

Comment:
Please update the test location as following:
```suggestion
        location = TEST_LOCATION
```